backend/libraries/ballerina-core-python/ballerina_core/parsing/option.py

```diff
@@ -33,8 +33,12 @@ def from_json(value: Json) -> Option[_Option]:
                         if _VALUE_KEY not in value:
                             raise ValueError(f"Missing value: {value}")
                         return Option.some(some_from_json(value[_VALUE_KEY]))
-                    case discriminator if discriminator == _NOTHING_VALUE:
-                        return Option.nothing()
+                    case discriminator if discriminator == _NONE_VALUE:
+                        if _VALUE_KEY not in value:
+                            raise ValueError(f"Missing value: {value}")
+                        if none_from_json(value[_VALUE_KEY]) != unit:
```

Comment:
this seems unnecessary

Reply:
Yes, I only need the side effect to ensure that the value is correct. Will fix in: https://github.com/Ballerina-Org/ballerina/pull/223